url6f0c51b1b4_TheFakeMontyOnTheRun_open688_tgz-pJ8-br_odb_open688_app_Open688ServerJ8/src/br/odb/open688/app/StatusTickerRunnable.java

```diff
@@ -0,0 +1,31 @@
+package br.odb.open688.app;
+
+import br.odb.open688.app.Open688Server.SimulationStatus;
+
+public class StatusTickerRunnable implements Runnable {
+
+	private Open688Server context;
+
+	public StatusTickerRunnable(Open688Server context) {
+		this.context = context;
+	}
+
+
+	@Override
+	public void run() {
+
+		while ( context.isConnected() ) {
+			
+			try {
+				Thread.sleep( 1000 );
+			} catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)